remove-retired-packages

```diff
@@ -45,26 +51,22 @@ echo "Asking for super user access:"
 sudo true Executed from remove-retired-packages
 
 echo "These packages have been retired:"
-for PACKAGE in $( cat "$TO_REMOVE"); do
-        #skip if this package is not installed
-        rpm -q "$PACKAGE" 2>/dev/null >&2 || continue
```

Comment:
I had to return this back. Without this - when A is dependency of B. And as first step you ask if remove of A is fine (but it will remove B as well). And as second step you as to remove B, but B is not already installed so all these rpm -q will fail.

Reply:
@xsuchy  I see. Perhaps it's sufficient to check the exit status of an rpm query command in the `dnf remove` loop and skip already removed dependencies there like this:

```
--- a/remove-retired-packages
+++ b/remove-retired-packages
@@ -59,9 +59,13 @@ while read PACKAGE; do
 done < "$RM_LIST"
 
 while read -u 7 PACKAGE; do
-        SUMMARY=$(rpm -q --qf "%{summary}" "$PACKAGE" | head -n1 )
+        SUMMARY="$(rpm -q --qf "%{sourcerpm}\n%{summary}" "$PACKAGE")"
+        if [ $? -eq 1 ]; then
+            continue
+        fi
+        SRPMFILENAME=$(echo "$SUMMARY" | head -n1)
+        SUMMARY=$(echo "$SUMMARY" | head -n2 | tail -n1)
         echo "Removing $PACKAGE: $SUMMARY"
-        SRPMFILENAME=$(rpm -q --qf "%{sourcerpm}" "$PACKAGE")
         SRPM=$(rpm-print-name-from-filename "$SRPMFILENAME")
         REASON=$(curl --fail https://src.fedoraproject.org/rpms/$SRPM/raw/rawhide/f/dead.package 2>/dev/null || echo "unknown reason")
         echo "Reason of retirement: $REASON"
```

That way we can keep avoiding to test every package from the TO_REMOVE list.

This also eliminates one `rpm` command invocation per retired package.